### PR TITLE
update giflib pin

### PR DESCRIPTION
**Pull request by ocefpaf**
Ping @isuruf and @CJ-Wright.

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -187,7 +187,7 @@ pin_run_as_build:
   geotiff:
     max_pin: x.x
   giflib:
-    max_pin: x.x
+    max_pin: x.x.x
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: These 2 lines are unnecessary if you mark the build number=0 packages as broken

Comment (ocefpaf) [reply recipe/conda_build_config.yaml]: You mean rely only on the `pin_subpackage` in the package and don't specify

```
giflib:
  max_pin: x.x.x
```

here? I'm OK with that.

Comment (isuruf) [reply recipe/conda_build_config.yaml]: yes

**Comment by CJ-Wright** [comment]
Doe we need make certain that all of our packages are built against the pin (eg new libraries could be only built against the newer number)?

**Comment by ocefpaf** [comment]
> Doe we need make certain that all of our packages are built against the pin (eg new libraries could be only built against the newer number)?

I guess that would be best. I don't think we have many packages that depend on `giflib` though. (I already rebuild `gdal`.)

**Comment by CJ-Wright** [comment]
Here are the successors:
```
['emacs',
 'imagemagick',
 'libgdal',
 'bob.io.image',
 'libwebp',
 'leptonica',
 'tesseract',
 'gdal',
 'root',
 'openimageio']
```

**Comment by ocefpaf** [comment]
@CJ-Wright I took the liberty of editing your message to keep track of the rebuilds.

Is this one good to be merged? I can issue re-render to those feedstocks. They are not that many. (And in theory they are already broken with the latest `giflib`.)

**Comment by CJ-Wright** [comment]
LGTM thank you very much @ocefpaf

**Pull request merged**